api/operator/v1beta1/vmalertmanager_types.go

```diff
@@ -562,8 +562,10 @@ func (cr *VMAlertmanager) SetUpdateStatusTo(ctx context.Context, r client.Client
 
 // AlertmanagerWebConfig defines web server configuration for alertmanager
 type AlertmanagerWebConfig struct {
-	// TLSServerConfig defines tls configuration for alertmanager web server
-	TLSServerConfig *WebserverTLSConfig `json:"tls_server_config,omitempty"`
+	// TLSServerConfig defines server TLS configuration for alertmanager
+	TLSServerConfig *TLSServerConfig `json:"tls_server_config,omitempty"`
+	// TLSClientConfig defines client TLS configuration for alertmanager
+	TLSClientConfig *TLSClientConfig `json:"tls_client_config,omitempty"`
```

Comment:
looks like there is no client configuration at web server level for alertmanager.

 Client configuration makes sense only for GOSSIP in-cluster communication.

It uses separate configuration file provided via `--cluster.tls-config` flag

https://prometheus.io/docs/alerting/latest/https/#gossip-traffic

I'd suggest to:
1) create new spec entry gossipServerConfig
2) add TLSServerConfig and TLSClientConfig to it.
3) conditionally and flag to the alertmanager configuration